src/useLastLocation.spec.tsx

```diff
@@ -1,137 +1,68 @@
-
-const suite = () => {
-  // We need to reset lastLocation for each run in tests
-  setLastLocation(null);
-
-  const appWrapper = mount(
-    <MemoryRouter initialEntries={['/']} initialIndex={0}>
-      <LastLocationProvider>
-        <div>
-          <Navigation />
-          <Route path="/about" component={About} />
-          <Route path="/contact" component={Contact} />
-          <Route path="/" component={Home} exact />
-        </div>
-      </LastLocationProvider>
-    </MemoryRouter>,
-  );
-
-  const getLink = (text: string) => appWrapper
-    .find(Link)
-    .find({ to: text })
-    .children();
+function renderHook() {
+  const history = createMemoryHistory({
+    initialEntries: ['/'],
+    initialIndex: 0,
+  });
 
   return {
-    get homeLink() {
-      return getLink('/');
-    },
-    get aboutLink() {
-      return getLink('/about');
-    },
-    get contactLink() {
-      return getLink('/contact');
-    },
-    get lastLocation() {
-      return appWrapper.find('.last-location-pathname');
-    },
-    get lastLocationPathname() {
-      return this.lastLocation.text();
-    },
-    get hasLastLocation() {
-      return this.lastLocation.length === 1;
-    },
-    get pageTitle() {
-      return appWrapper.find('h1').text();
-    },
-    click(wrapper: ReactWrapper) {
-      // We need to pass mock event { button: 0 }
-      // @see https://github.com/airbnb/enzyme/issues/516#issue-167924470
-      wrapper.simulate('click', { button: 0 });
-    },
+    history,
+    ...rtlRenderHook(() => useLastLocation(), {
+      wrapper: ({ children }) => (
+        <Router history={history}>
+          <LastLocationProvider>
+            {children}
+          </LastLocationProvider>
+        </Router>
+      ),
+    }),
   };
-};
+}
 
 describe('useLastLocation', () => {
-  // This one is a bit silly but I wanted to be 100% sure that
-  // my mini react application in tests is working correctly
-  it('should change content when route changes', () => {
-    const $ = suite();
-    expect($.pageTitle).toBe('Home');
-
-    $.click($.aboutLink);
-    expect($.pageTitle).toBe('About');
-
-    $.click($.contactLink);
-    expect($.pageTitle).toBe('Contact');
-  });
+  it('should have no last location on the first visit', async () => {
+    const { result } = renderHook();
 
-  it('should not render last location on the first visit', () => {
-    const $ = suite();
-    expect($.hasLastLocation).toBe(false);
+    expect(result.current).toBe(null);
   });
 
   it('Home ► About, should show / as last location', () => {
-    const $ = suite();
-    expect($.hasLastLocation).toBe(false);
+    const { history, result } = renderHook();
+    expect(result.current).toBe(null);
 
-    $.click($.aboutLink);
-    expect($.hasLastLocation).toBe(true);
-    expect($.lastLocationPathname).toBe('/');
+    act(() => {
+      history.push('/about');
+    });
+
+    expect(result.current).not.toBe(null);
+    // @ts-ignore
```

Comment:
why does this need to be ignored?

Reply:
VS Code was kind-of contradicting itself with "result.current could be null". But then it wouldn't let me optional chain it (i.e. `result.current?.pathname`)